File: ch12_the_sparse_extended_information_filter/src/matlab/equation_measurement.m
% ==============================================================

%equations of measurement model
%written by Pierre-Paul TACHER (ptacher@gmail.com)

%m:current mean estimate
%c: correspondance vector

%F: measurement model 2 x n

% =============================================================
function [ F ] = equation_measurement(m,c)

    n=size(c,2);
    c1 = 2 + 2*c';
    x = m(c1);y = m(c1+1);

    delta = bsxfun(@minus ,[x';y'], m(1:2));
    F = [sqrt(sum(delta.^2));zeros(1,n)];
    F(2,:) = atan2(delta(2,:),delta(1,:))-m(3)+pi/2;
    
end

File: ch12_the_sparse_extended_information_filter/src/python/equation_measurement.py
from numpy import *

def equation_measurement(m,ind):
    N = ind.size
    ind1 = 1+2*ind
    x = m[ind1]
    y = m[ind1+1]

    delta = vstack((x,y))-m[0:2].reshape((2,1))
    q = sqrt(sum(delta**2,axis=0))
    return vstack((q,arctan2(delta[1,:],delta[0,:])-m[2]+pi/2))
